src/tools/timeseries-generation/main.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#include <memory>
+#include <string>
+
+#include <antares/exception/LoadingError.hpp>
+#include <antares/study/header.h>
+#include <antares/study/study.h>
+#include <antares/logs/logs.h>
+#include <yuni/core/getopt.h>
+
+#include <antares/timeseries-generation/timeseries-generation.h>
+
+using namespace Antares;
+
+std::unique_ptr<Yuni::GetOpt::Parser> createTsGeneratorParser(TsGeneratorSettings& settings)
```

Comment:
nice! In order to avoid Yuni in new codes there is [boost program options](https://www.boost.org/doc/libs/1_63_0/doc/html/program_options.html)

Reply:
We'll migrate all parsers at the same time, when removing the dependency to Yuni

Reply:
We'll use an alternativer to yuni parser in a later PR
https://github.com/AntaresSimulatorTeam/Antares_Simulator/issues/1819